projects/rocprofiler-sdk/source/bin/rocprofv3.py

```diff
@@ -437,6 +437,44 @@ def add_parser_bool_argument(gparser, *args, **kwargs):
         nargs="*",
     )
 
+    spm_options = parser.add_argument_group("Streaming Perfmon options")
+
+    add_parser_bool_argument(
+        spm_options,
+        "--spm-beta-enabled",
+        help="enable SPM; beta version",
+    )
+
+    spm_options.add_argument(
+        "--spm",
+        help=(
+            "Specify SPM events to collect(comma OR space separated in case of more than 1 counters). "
+            "Note: job will fail if entire set of counters cannot be collected in single pass"
+        ),
+        default=None,
+        nargs="*",
+    )
+
+    spm_options.add_argument(
+        "--spm-buffer-size",
+        help="SPM Buffer size in kilobytes. Default value is set to 32768 KB in tool",
+        default=None,
+        type=str,
+    )
+
+    spm_options.add_argument(
+        "--spm-timeout",
+        help="Timeout for SPM, in ms. Larger values may slow down the application, while smaller values may drop data. Default value is set to 50 ms in tool",
+        default=None,
+        type=int,
+    )
+
+    spm_options.add_argument(
+        "--spm-frequency",
+        help="Frequency in Mhz. This is estimated to shader clock count. Default is set to 500000 in tool.",
```

Comment:
Do we maybe want this to be GHz? 500,000 as a default value is awfully large. Can you specify a frequency < 1 GHz? And if so, we could still make this GHz and support decimal values, e.g. `--spm-frequency 0.5` would be 500 MHz.